packages/snaps.sls

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+# vim: ft=sls
+{% from "packages/map.jinja" import packages with context %}
+
+{% set req_states = packages.snaps.required.states %}
+{% set wanted_snaps = packages.snaps.wanted %}
+{% set unwanted_snaps = packages.snaps.unwanted %}
+
+### REQ PKGS (without these, some WANTED SNAPS fail to install/uninstall)
+include:
+  - packages.pkgs
+
+extend: 
+  unwanted_pkgs:
+    pkg.removed:
+      - pkgs: {{ packages.snaps.collides }}
```

Comment:
As `packages.snaps.collides` is always set in `defaults.yml`, some packages will be always removed even if you don't want to manage snap packages.  This will happen, at least with `snap` (see above)

Perhaps we can keep all these packages removals conditioned to `packages.snaps.wanted != []` or `packages.snaps.unwanted != []`?

Reply:
I'm happy to change this - if you clearly see a nice solution - pls suggest.  

So should I wrap extend inside `jinja` do you think?  or maybe onlyif/unless.


(BTW: Normally `extends` goes to `init.sls` where includes are centralized but one thing at a time).

Reply:
`onlyif/unless`  are 'cleaner' to me, but in this case, as there are 2 or'ed conditions and a few places where to use them, perhaps an `if` block surrounding [all these](https://github.com/noelmcloughlin/packages-formula/blob/2a0a38009e9b074766f765efddc8fe93977377cf/packages/snaps.sls#L13L41) will do for now?

Reply:
@javierbertoli   Thanks for the review.  

I agree. I've wrapped block with Jinja2 and I think the comparison operators are not needed -  Lets see if Travis CI works and I test again on Fedora shortly.

Reply:
I tested `packages`  and `packages.pkgs, gems, pips` (i.e. without snapd) on Fedora and both high-states were successful.

Reply:
Travis CI was successful - Issue #14 looks like noise on Debian.